cabal-install/Distribution/Client/ProjectPlanning.hs

```diff
@@ -1583,24 +1584,30 @@ elaborateInstallPlan verbosity platform compiler compilerprogdb pkgConfigDB
                          -> SolverId -> [ElaboratedPlanPackage]
     elaborateLibSolverId mapDep = filter (matchPlanPkg (== CLibName)) . mapDep
 
-    -- | Given an 'ElaboratedPlanPackage', return the path to where the
-    -- executable that this package represents would be installed.
-    planPackageExePath :: ElaboratedPlanPackage -> Maybe FilePath
-    planPackageExePath =
+    -- | Given an 'ElaboratedPlanPackage', return the paths to where the
+    -- executables that this package represents would be installed.
+    -- The only case where multiple paths can be returned is the inplace
+    -- monolithic package one, since there can be multiple exes and each one
+    -- has its own directory.
+    planPackageExePaths :: ElaboratedPlanPackage -> [FilePath]
+    planPackageExePaths =
         -- Pre-existing executables are assumed to be in PATH
         -- already.  In fact, this should be impossible.
-        InstallPlan.foldPlanPackage (const Nothing) $ \elab -> Just $
+        InstallPlan.foldPlanPackage (const []) $ \elab ->
             binDirectoryFor
                 distDirLayout
                 elaboratedSharedConfig
-                elab $
+                elab <$>
```

Comment:
I'd use `map` instead of `<$>` here as well.

Reply:
Like `map (binDirectoryFor ...) case ...`? or inline `map`? The first is a bit hard to follow imo, and inline map is just `<$>` (it's clear that we are acting on a `[]` from the type sig)... maybe a comment instead?

Or what about a binding?
```suggestion
                elab
                <$> executables
        where
            executables =
```

Reply:
I was thinking of `` `map` `` instead of `<$>`.